### DISCUSSION TITLE
nnUNet region based training not working as intended

### DISCUSSION THREAD

**Discussion by iffthomas** [Project-MONAI/MONAI#7237] [Q&A]
Hey,

I've been running the nn u-netmodel for images with corresponding segmentation masks which have 19 classes.
Individually classifying them lead to the results whats has been reported in the literature. As these 19 classes can be grouped to three different groups I've tried region based training. All Classes don't overlap

I've constructed my json file as follows:
S and V result in a fairly good score, however for the 3rd group: "D" the Dice Score is zero.
For the first model where I trained on the individual classses I've used the weighted average to calculate the group average Dice Score.

Throughout the Training process for region bases training I get PseudoDice scores about 0.9 for all groups  and in the the validation will drop to 0.5 for 2 groups "V" and "D" 

I've constructed my json file as follows:
Did I do a mistake in my json file
or could the reason be that a for "D" not all classes are present but this argument would be not be very strong as also some classes are missing in "V" for some images just not as much for "V" as for "D"

{
"name": "Dataset004,
"channel_names": {
"both": "0"
},
"labels": {
"background": 0,
"S": 1,
"V": [2,3,4,5,6,7,8,9,10],
"D": [11,12,13,14,15,16,17,18,19]
},
"regions_class_order":[1,2,3],
"numTraining": 406,
"file_ending": ".nii.gz",
"overwrite_image_reader_writer": "NibabelIO"
}

"mean": {
    "(11, 12, 13, 14, 15, 16, 17, 18, 19)": {
        "Dice": 0.0,
        "FN": 115488.97044334975,
        "FP": 0.0,
        "IoU": 0.0,
        "TN": 66993375.02955665,
        "TP": 0.0,
        "n_pred": 0.0,
        "n_ref": 115488.97044334975
    },
    "(2, 3, 4, 5, 6, 7, 8, 9, 10)": {
        "Dice": 0.8317142591187485,
        "FN": 50924.54926108375,
        "FP": 163123.11083743843,
        "IoU": 0.7124195923968052,
        "TN": 66363863.2955665,
        "TP": 530953.0443349754,
        "n_pred": 694076.1551724138,
        "n_ref": 581877.5935960591
    },
    "1": {
        "Dice": 0.903534736077213,
        "FN": 9775.573891625616,
        "FP": 10517.926108374384,
        "IoU": 0.8246761041694668,
        "TN": 66992020.39162561,
        "TP": 96550.10837438423,
        "n_pred": 107068.03448275862,
        "n_ref": 106325.68226600986
    }

**Comment by KumoLiu** [comment]
Hi @iffthomas, thanks for your interest here.

[NnUNet's docs](https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/region_based_training.md) mention that "Region-based training can be used to learn overlapping labels", not sure whether it will be the issue. But you could simply combine your labels and try to use regular training. Thanks!
